src/contracts/core/StrategyManager.sol

```diff
@@ -310,13 +324,18 @@ contract StrategyManager is
         token.safeTransferFrom(msg.sender, address(strategy), amount);
 
         // deposit the assets into the specified strategy and get the equivalent amount of shares in that strategy
-        shares = strategy.deposit(token, amount);
+        uint256 shares = strategy.deposit(token, amount);
 
         // add the returned shares to the staker's existing shares for this strategy
-        _addShares(staker, token, strategy, shares);
+        uint256 existingShares = _addShares(staker, token, strategy, shares);
```

Comment:
instead of calling `addShares` on completion of withdrawal as shares, i think we should combine `_addShares` and `increaseDelegatedShares` into an `increaseStakedShares` that is called so it is as close to deposit processing as possible

Reply:
Same thing in SM